### ISSUE TITLE
Steam store shows games not available for Linux (Big Picture mode + web)

### ISSUE THREAD

**Issue by Cybolic** [ValveSoftware/steam-for-linux#4074]
Even though the setting Store->"Let me view all content, regardless of my current setup" is set to off and don't have any Windows machines on my network or available, the store still pretty much only shows Windows games; this also happens on the website.

Possibly related issues:

+ https://github.com/ValveSoftware/steam-for-linux/issues/3929
+ https://github.com/ValveSoftware/steam-for-linux/issues/3999

![steam-store-pref-only-linux](https://cloud.githubusercontent.com/assets/1196197/10604536/9ae7de0e-772d-11e5-897c-c3985d2e9ffd.jpg)

**Comment by liamdawe** [comment]
This has been happening for quite a while. It's especially annoying on the web store to see "recommended for you" to be always Windows games when on Linux.

It used to show Linux games only for me, and suddenly it doesn't. It's completely pointless to show Windows games in the features areas when not on Windows.

**Comment by Seegras** [comment]
Yes, the web page now for some reason defaults to Windows, when on Linux, and if I click on All Specials or All New Releases, it shows me the [x] Windows pre-selected. It's been that for a few months, before that it worked as expected.

**Comment by adolson** [comment]
This would be a great bug to fix between now and Tuesday, when Steam Machines head out to retail...

**Comment by kon14** [comment]
This used to work as expected, please fix it, it's November the 7th already.

**Comment by Mblackwell1024** [comment]
The web store is definitely broken/not filtered (and the option isn't even there) but on Ubuntu 15.04 running the latest client beta, I only have Linux games in Desktop and Big Picture mode. However it's worth noting that in desktop mode under Customize "Available for Linux/SteamOS" is checked.

Could this be a SteamOS specific problem?

**Comment by liamdawe** [comment]
On SteamOS the in-store option is checked by default, but it still lists Windows games.

If I uncheck it, then check it again it will now only show me SteamOS games. So it seems like the UI is checked, but the query behind it isn't doing the check for some reason until you uncheck and recheck it.

**Comment by adolson** [comment]
With the latest beta, I don't see the option in the Settings menu anymore. I can disable and then enable the "only on my current OS" option inside the Store itself. However, this doesn't affect the main categories, such as Featured, which still shows Windows-only games such as Act of Aggression and Age of Empires 2 HD...

**Comment by xpander69** [comment]
on the webpage i still see windows games by browsing store.steampowered.com
browser detection is broke? it happened i think 1 month ago
tested browsers: Vivaldi, Opera, Chromium, Firefox.

its really nasty to browse in the store like that.
please fix it

**Comment by Mblackwell1024** [comment]
For those commenting about the available games not being correct on the web, and people looking into it, it's worth noting this is the default string in Ubuntu w/ Firefox 42:
Mozilla/5.0 (X11; Ubuntu; Linux x86_64; rv:42.0) Gecko/20100101 Firefox/42.0

And this is the agent in the steam client (at least what I saved a few weeks ago):
Mozilla/5.0 (Linux; U; X11; en-US; Valve Steam Client/1440016726; ) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/43.0.2357.81 Safari/537.36

Note that the only part that matters is the Steam Client section. After some testing even just adding "Valve Steam Client;" to the default user agent string did the trick. So apparently whatever checking the store page does for OS skips over everything else.

**Comment by calexil** [comment]
yeah thats gonna be a big problem... I don't wanna have to tell every fucking steamos noob they cant play gta v despite it showing up

**Comment by Cybolic** [comment]
A related problem has appeared regarding user agent strings; if "Valve Steam Client" is in the string, then it's no longer possible to log-in or have access to purchase, follow and related options for products, making the user agent switch unusable for anything other than browsing the main page.

**Comment by xpander69** [comment]
and it still hasnt been fixed... tired of going through all the games on the store that aren't even available on my platform, let alone the confusing steamOS logo

**Comment by adolson** [comment]
Valve, please fix this!

Alternately, please convince every dev to port every game so this becomes a
non-issue.

**Comment by Seegras** [comment]
Maybe the issue should be renamed to something like "Webstore and Big Picture: OS and Browser detection broken for Linux".

**Comment by not83** [comment]
Wasn't this fixed a while ago?

**Comment by adolson** [comment]
It is mostly fixed, yes.

It still shows non-Linux games if you browse under the Your Store > For You section. Unless DiRT Rally, Path of Exile, Fallout 4, The Witcher 3, Castle Crashers, etc are all suddenly on Linux...

At first I thought it was maybe some caching thing, because once you go into a tag there, like say, Singleplayer, it will default to "Most Popular" and will show Windows games. Then if you go to "Top Rated" there won't be Windows games, and if you go *back* to "Most Popular", suddenly the Windows games are gone. But if you back out of that screen, and go back in, the Windows games are back again. So, there's still a bug there at least for me. And I'm using the latest beta (I updated right before posting this comment).

It would also be nice if the Wishlist would filter out non-Linux games, depending on the filter setting we chose. I watch some games for ports by wishlisting them, but don't need to see them in Steam Big Picture's Store until they're ported. Either that, or at least show more than 40 games from my wishlist at a time so I can easily get to ones that are available for purchase.

**Comment by not83** [comment]
Those games ARE under Linux aren't they?

**Comment by adolson** [comment]
Are you seriously asking that?

**Comment by not83** [comment]
Oh thought that said witcher 2. Isn't dirt rally on Linux however

**Comment by adolson** [comment]
No..

**Comment by Cybolic** [comment]
It's still not fixed at all using the website.

**Comment by Seegras** [comment]
It's not fixed on the webpage. "Recommended for you" on the front page still shows various platforms. And also the tabs "Popular New Releases", "Top Sellers", "Upcoming", "Specials" constantly show various platforms. If you click on "All ..." to see all the games in the above respective categories, the list comes with Windows preselected, on a browser that's obviously NOT windows: Mozilla/5.0 (X11; Linux x86_64; rv:44.0) Gecko/20100101 Firefox/44.0 Iceweasel/44.0 So you have to first select "SteamOS + Linux" and then unselect "Windows".

**Comment by xpander69** [comment]
and its still not fixed. 
works in the steam client, but using every browser (firefox, chrome, opera, vivaldi, midori ) still displaying Windows Games in store.

Come on, fix it already, i dont want to browse windows games

**Comment by diego-XA** [comment]
This has changed for me too about a couple of months ago and now I barely visit the store because it is very frustrating.

**Comment by Swiftpaw** [comment]
This is now happening for me in the tabbed section at the bottom. All tabs include games that aren't on Linux yet, and the "Popular New Releases" and "All New Releases" buttons both take me to a search page with Linux unchecked, and both Mac and Windows checked, the exact opposite of what it should be.

**Comment by diego-XA** [comment]
> All tabs include games that aren't on Linux yet, and the "Popular New Releases" and "All New Releases" buttons both take me to a search page with Linux unchecked, and both Mac and Windows checked, the exact opposite of what it should be.

That is exactly what is happening to me too.

**Comment by ediazfer** [comment]
Same here.  I'm not checking the store anymore.  Having to filter-out all the non-Linux software takes too much time.
And using "search" is not the same as having the new/hot products on the front page.

**Comment by ediazfer** [comment]
Same here. I'm not checking the store anymore. Having to filter-out all the non-Linux software takes too much time.
And using "search" is not the same as having the new/hot products on the front page.

**Comment by Swiftpaw** [comment]
This issue is now mostly fixed.  The only remaining sections that don't respect your platform preference filter selection are the "Upcoming" tab, the Daily Deals, and the big promos.  Not sure if the Weekly Deal ever does it but so far it hasn't yet for me.

**Comment by waltercool** [comment]
On Steam autumn sale, I saw a lot of non-linux games, even if my configuration specify ONLY Linux games. This is still broken

**Comment by waltercool** [comment]
As workaround, I been blacklisting games, but that's not a real solution, not even for game developers who might want to release the game for a specific platform in future, but the listing would be banned.

**Comment by Swiftpaw** [comment]
Yep, special sale pages such as the current store featured page show a lot of games that aren't respecting platform filter options.  Makes the featured page useless as trying to look through them is too annoying.

**Comment by ediazfer** [comment]
Exactly that.  I looked at the Black Friday sales, started to check for the ones actually available to Linux, gave up after a handful.
The hundreds of games I own on steam have been bought because it's easy to see them and buy them.
Not because I really planned to play with them.

**Comment by aaronfranke** [comment]
There is an option at the bottom of [account preferences](https://store.steampowered.com/account/preferences/) to set your platform to SteamOS + Linux

Despite this setting being checked, the desktop Steam Client Store page, the Steam website, and the Store in Steam Big Picture mode all still show non-Linux games.
